backend/pkg/helm/release.go

```diff
@@ -0,0 +1,763 @@
+	_, err = h.Configuration.Releases.Deployed(req.Name)
+	if err == driver.ErrReleaseNotFound {
+		zlog.Error().Err(err).Str("releaseName", req.Name).
+			Str("request", "get_release_history").Msg("release not found")
+		http.Error(w, err.Error(), http.StatusNotFound)
+
+		return
+	}
+
+	getClient := action.NewHistory(h.Configuration)
+
+	result, err := getClient.Run(req.Name)
+	if err != nil {
+		zlog.
+			Error().
+			Err(err).
+			Str("request", "get_release_history").
+			Str("releaseName", req.Name).
+			Msg("failed to get release history")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+
+		return
+	}
+
+	resp := GetReleaseHistoryResponse{
+		Releases: result,
+	}
+
+	w.WriteHeader(http.StatusOK)
+
+	err = json.NewEncoder(w).Encode(resp)
+
+	if err != nil {
+		zlog.
+			Error().
+			Err(err).
+			Str("request", "get_release_history").
+			Str("releaseName", req.Name).
+			Msg("failed to encode response")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+}
+
+type UninstallReleaseRequest struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+func (h *Handler) UninstallRelease(w http.ResponseWriter, r *http.Request) {
+	// Parse request
+	var req UninstallReleaseRequest
+
+	decoder := schema.NewDecoder()
+
+	err := decoder.Decode(&req, r.URL.Query())
+	if err != nil {
+		zlog.Error().Err(err).Str("action", "uninstall").Msg("failed to parse request")
+		http.Error(w, err.Error(), http.StatusBadRequest)
+
+		return
+	}
+
+	// check if release exists
+	_, err = h.Configuration.Releases.Deployed(req.Name)
+	if err == driver.ErrReleaseNotFound {
+		zlog.Error().Err(err).Str("releaseName", req.Name).
+			Str("action", "uninstall").Msg("release not found")
+		http.Error(w, err.Error(), http.StatusNotFound)
+
+		return
+	}
+
+	go func(h *Handler) {
+		actionState.SetStatus("uninstall", req.Name, "processing", nil)
+		h.uninstallRelease(req)
+	}(h)
+
+	response := map[string]string{
+		"message": "uninstall request accepted",
+	}
+
+	w.WriteHeader(http.StatusAccepted)
+
+	err = json.NewEncoder(w).Encode(response)
+
+	if err != nil {
+		zlog.Error().Err(err).Str("action", "uninstall").Msg("failed to encode response")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+}
+
+func (h *Handler) uninstallRelease(req UninstallReleaseRequest) {
+	// Get uninstall client
+	uninstallClient := action.NewUninstall(h.Configuration)
+
+	_, err := uninstallClient.Run(req.Name)
+	if err != nil {
+		zlog.Error().Err(err).Str("releaseName", req.Name).Str("namespace", req.Namespace).Msg("failed to uninstall release")
+		actionState.SetStatus("uninstall", req.Name, "failed", err)
+
+		return
+	}
+
+	actionState.SetStatus("uninstall", req.Name, "success", err)
+}
+
+type RollbackReleaseRequest struct {
+	Name      string `json:"name" validate:"required"`
```

Comment:
I wonder if these fields need to be validated further by us, or if the helm code does more validation?